docs/spec/spec.md

```diff
@@ -0,0 +1,314 @@
+
+## 1. Overview
+
+The Ballerina SAP JCo Connector provides seamless integration with SAP systems through Java Connector (JCo) capabilities. This connector allows Ballerina applications to interact with SAP systems, enabling operations like Remote Function Call (RFC) execution, IDoc processing, and more. By leveraging this connector, developers can easily integrate SAP functionalities into their Ballerina applications, making it an essential tool for enterprises working with SAP.
+
+The `ballerinax/sap.jco` package exposes the SAP JCo library as Ballerina functions, enabling easy access to SAP's enterprise resource planning (ERP) software and other SAP solutions, such as human capital management (HCM), customer relationship management (CRM), and supply chain management (SCM).
+
+### Architecture Overview
+
+The SAP JCo Connector architecture comprises key components facilitating communication between Ballerina applications and SAP systems. These components include the SAP function modules, the SAP system itself, the Ballerina code that handles the RFC calls, and the IDoc servers responsible for sending and receiving IDocs. This architecture enables robust integration, supporting various enterprise operations.
+
+## 2. Components
+
+This section outlines the core components of the Ballerina SAP JCo Connector and their configurations, focusing on how they interact with the SAP system.
+
+### 2.1 RFC Client
+
+The RFC Client component allows the Ballerina application to communicate with an SAP system by invoking remote function modules and sending IDocs.
+
+#### 2.1.1 Configurations
+
+The RFC Client supports two types of configurations, `DestinationConfig` and `AdvancedConfig`, to establish a connection with the SAP system.
+
+##### 2.1.1.1 DestinationConfig
+
+The `DestinationConfig` type represents the configuration details needed to create a RFC connection.
+
+```ballerina
+public type DestinationConfig record {]
+    # The SAP hostname (jco.client.ashost).
+    string ashost;
+    # The SAP system number (jco.client.sysnr).
+    string sysnr;
+    # The SAP client (jco.client.client).
+    string jcoClient;
+    # The SAP user name (jco.client.user).
+    string user;
+    # The SAP password (jco.client.passwd).
+    string passwd;
+    # The SAP language (jco.client.lang).
+    string lang = "EN";
+    # The SAP group (jco.client.group).
+    string group = "PUBLIC";
+};
+```
+
+**Example Configuration:**
+
+```ballerina
+configurable jco:DestinationConfig config = ?;
+jco:Client sapClient = check new (config);
+```
+
+Config.toml file:
+
+```toml
+[config]
+ashost = "host"
+sysnr = "00"
+jcoClient = "000"
+user = "user"
+passwd = "password"
+lang = "en"
+group = "group"
+```
+
+##### 2.1.1.2 AdvancedConfig
+
+If the user wants to specify more detailed configurations beyond `DestinationConfig`, they can use `AdvancedConfig`. The `AdvancedConfig` type is a map that holds any SAP JCo configurations accepted by the SAP destination provider. 
```

Comment:
Can you include the `AdvancedConfig` record definition as well ?

Reply:
Included with [5fb5481](https://github.com/ballerina-platform/module-ballerinax-sap.jco/pull/56/commits/5fb5481cd4cd3236393eb4704d2c54fd71908db8).